src/update_engine/payload_processor.cc

```diff
@@ -145,35 +173,60 @@ ActionExitCode PayloadProcessor::LoadManifest() {
 ActionExitCode PayloadProcessor::PerformOperation() {
   DCHECK(next_operation_num_ < num_total_operations_);
 
-  const InstallOperation &op =
-      manifest_.partition_operations(next_operation_num_);
+  const size_t total_partition_operations = manifest_.partition_operations_size();
+  const InstallOperation *op = nullptr;
+  DeltaPerformer *performer = nullptr;
 
-  if (op.data_length() && op.data_offset() != buffer_offset_) {
+  if (next_operation_num_ < total_partition_operations) {
+    op = &manifest_.partition_operations(next_operation_num_);
+    performer = &partition_performer_;
+  } else {
+    size_t proc_operation = next_operation_num_ - total_partition_operations;
+
+    for (const InstallProcedure &proc : manifest_.procedures()) {
```

Comment:
Could you be more specific? `proc_operation` is an index of the overall set of operations, but the operations are essentially in a list of lists of varying sizes so it isn't useful directly.

Part of this awkward is inherited from the existing code structure of tracking `next_operation_num_` in the first place. I could instead drop next_operation_num_ and instead keep lists of pending procedures and the current procedure's operations and just keep popping them off as needed. That'd probably look a bit cleaner.